src/parse/localization.rs

```diff
@@ -438,28 +438,33 @@ impl<'a> ValueParser<'a> {
         }
     }
 
-    fn peek(&mut self) -> Option<char> {
-        let p = self.content_iters[self.content_idx].peek();
-        if p.is_none() {
-            if self.content_idx + 1 == self.content.len() {
-                None
-            } else {
-                self.content_idx += 1;
-                self.loc = self.content[self.content_idx].loc;
-                self.offset = 0;
-                self.peek()
-            }
+    fn maybe_advance_idx(&mut self) -> bool {
+        if self.content_idx + 1 == self.content.len() {
+            false
         } else {
-            p.copied()
+            self.content_idx += 1;
+            self.loc = self.content[self.content_idx].loc;
+            self.offset = 0;
+            true
+        }
+    }
+
+    fn peek_or_advance_idx(&mut self) -> Option<char> {
```

Comment:
I don't get this name change. `self.peek()` always peeks. Under the hood, it advances content_idx as necessary to do the peek, but that doesn't change what it does. There's no "or" here. After calling `maybe_advance_idx`, it recurses to ensure a peek is done.